plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -144,18 +137,15 @@ public PinotMetadata(
     public PinotTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
     {
         if (tableName.getTableName().trim().startsWith("select ")) {
-            DynamicTable dynamicTable = DynamicTableBuilder.buildFromPql(this, tableName);
+            DynamicTable dynamicTable = DynamicTableBuilder.buildFromPql(this, pinotClient, tableName);
             return new PinotTableHandle(tableName.getSchemaName(), dynamicTable.getTableName(), TupleDomain.all(), OptionalLong.empty(), Optional.of(dynamicTable));
         }
 
-        try {
-            String pinotTableName = getPinotTableNameFromTrinoTableName(tableName.getTableName());
-            return new PinotTableHandle(tableName.getSchemaName(), pinotTableName);
-        }
-        catch (TableNotFoundException e) {
-            log.debug(e, "Table not found: %s", tableName);
+        String pinotTableName = pinotClient.getPinotTableNameFromTrinoTableNameIfExists(tableName.getTableName());
```

Comment:
nit: It might be helpful to use a name like `toRemoteTableName` and `toRemoteSchemaName` since many other connectors use those method names and it'll make it easier for readers to figure out what this is trying to do.

I don't have a strong opinion on this though.